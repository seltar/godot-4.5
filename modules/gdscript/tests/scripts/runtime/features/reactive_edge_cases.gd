extends RefCounted

@reactive
var counter: int = 0

@reactive
var message: String = "start"

func test():
	@warning_ignore("return_value_discarded")
	counter_changed.connect(_on_counter_changed)
	@warning_ignore("return_value_discarded")
	message_changed.connect(_on_message_changed)
	
	print("Testing multiple assignments:")
	
	# Test rapid successive changes
	counter = 1
	counter = 2
	counter = 3
	
	message = "middle"
	message = "end"
	
	# Test same value assignment (should not trigger)
	print("Assigning same value (should not trigger signal):")
	counter = 3  # Same as current value
	message = "end"  # Same as current value
	
	# Test null/empty assignments
	message = ""
	counter = 0

func _on_counter_changed(old_value, new_value):
	print("Counter: ", old_value, " -> ", new_value)

func _on_message_changed(old_value, new_value):
	print("Message: '", old_value, "' -> '", new_value, "'")
