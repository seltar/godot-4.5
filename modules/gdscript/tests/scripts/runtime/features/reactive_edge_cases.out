GDTEST_OK
Testing multiple assignments:
Counter: 0 -> 1
Counter: 1 -> 2
Counter: 2 -> 3
Message: 'start' -> 'middle'
Message: 'middle' -> 'end'
Assigning same value (should not trigger signal):
Message: 'end' -> ''
Counter: 3 -> 0
